source/extensions/common/wasm/wasm.cc

```diff
@@ -1722,21 +1722,38 @@ const StreamInfo::StreamInfo* Context::getRequestStreamInfo() const {
     return &decoder_callbacks_->streamInfo();
   } else if (access_log_stream_info_) {
     return access_log_stream_info_;
+  } else if (network_read_filter_callbacks_) {
+    return &network_read_filter_callbacks_->connection().streamInfo();
+  } else if (network_write_filter_callbacks_) {
+    return &network_write_filter_callbacks_->connection().streamInfo();
   }
   return nullptr;
 }
 
+// FilterState
+StreamInfo::FilterState& Context::getFilterState() const {
+  if (encoder_callbacks_) {
+    return encoder_callbacks_->streamInfo().filterState();
+  }
+  if (decoder_callbacks_) {
+    return decoder_callbacks_->streamInfo().filterState();
+  }
+  if (network_read_filter_callbacks_) {
+    return network_read_filter_callbacks_->connection().streamInfo().filterState();
+  }
+  return network_write_filter_callbacks_->connection().streamInfo().filterState();
+}
+
 WasmResult Context::setProperty(absl::string_view key, absl::string_view serialized_value) {
   ProtobufWkt::Value value_struct;
   if (!value_struct.ParseFromArray(serialized_value.data(), serialized_value.size())) {
     return WasmResult::ParseFailure;
   }
-  if (decoder_callbacks_ == nullptr && encoder_callbacks_ == nullptr) {
```

Comment:
Everything below could be simplified to:
```
auto stream_info = getRequestStreamInfo();
if (!stream_info) {
  return WasmResult:: NotFound;
}

stream_info->filterState().setData(key, std::make_unique<WasmState>(value_struct),
                                   StreamInfo::FilterState::StateType::Mutable);
return WasmResult::Ok;
```
and then you don't even need `getFilterState()`.

Reply:
I totally agree with that, but I kept separate functions because getStreamInfo was looking at access_log_stream_info_  and getFilterstate was not..  Is it okay to look at access_log_stream_info_ for filter state?